pkg/jxfactory/interface.go

```diff
@@ -0,0 +1,38 @@
+package jxfactory
+
+import (
+	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+
+	// this is so that we load the auth plugins so we can connect to, say, GCP
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+// Factory is the interface defined for Kubernetes, Jenkins X, and Tekton REST APIs
+//go:generate pegomock generate github.com/jenkins-x/jx/pkg/jxfactory Factory -o mocks/factory.go
```

Comment:
Interesting that the lint task passed, since the pegomocks are not part of the pull request. I think the problem is in the Makefile and _verify-generation-complete_ rule:

```Makefile
	$(eval CHANGED = $(shell git ls-files --modified -- . ':(exclude)pkg/client/openapi/k8s_io_apimachinery_intstr_unversioned/openapi_generated.go'))
```

It only bails on modified files. One probably needs to add `--others` as well to handle non tracked file:

```Makefile
	$(eval CHANGED = $(shell git ls-files --modified --others -- . ':(exclude)pkg/client/openapi/k8s_io_apimachinery_intstr_unversioned/openapi_generated.go'))
```

Reply:
I've checked in the generated mocks now